### PR TITLE
feature 7 course creation + course enrolment + tasks to tasklist use cases

### PR DESCRIPTION
**Pull request by jltng**
There is a lot of things going on even more things that need to be fixed!

Added a new CourseMap class in the entities layer that will be storing the courses.

1. **course_creation_use_case**
At the moment I have the RequestModel and screen such that the instructor creates the Course object by filling in screen (will need to somehow initialize the students arraylist)
Screen Shot 2022-11-19 at 12 48 16 PM

For task creation:
- Currently have a 'task name' and 'task id' on screen (task id should be unique and not up to instructor to create, will change it so that the id is the localdatetime created or something else that is unique)
- Then, would need to take those 2 arguments and create a new Task with that. 
- Then, would need to take task id and at it to the tasks arraylist argument of the Course entity in CourseMap

2. **CourseEnrolmentInteractor**
Need to still add student id to Course object’s list of students in CourseMap, save and update, and then send success view to presenter

3. **tasks_to_tasklist_use_case**
Needs to be implemented, one of the TAs said that since it's returning a simple data structure (only a succcess view message) there is a particular way to do it .... I shall ask them during the week to figure it out

4. **test cases:**
to be finished

### CODE REVIEW

File: src/main/java/course_enrolment_use_case/CourseEnrolmentRequestModel.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    private final String courseID;
+
+    public CourseEnrolmentRequestModel(String courseName, String courseInstructor) {
+        this.courseName = courseName;
+        this.courseInstructor = courseInstructor;
+        this.courseID = courseName + courseInstructor;
+    }
+
+    /**
+     * Change getters to public?
```
Comment (yyutiw) [inline src/main/java/course_enrolment_use_case/CourseEnrolmentRequestModel.java:17]: yes?

**Review by yyutiw — COMMENTED** [review]
Coding style: Class, package, and variable names all follow proper naming conventions and clearly describe their purpose (thank you for changing my entities package to lowercase). 

Clean arch: Implementation of use cases follow clean architecture and SOLID principles very well

Javadoc: Good documentation of code, each step is mostly labeled. However, documentation is missing for most methods/constructors.

**Review by yyutiw — APPROVED** [review]

**Review by TiareMar — APPROVED** [review]
No major problems in code, although testing and java documentation still need to be completed.

Good packaging organization and naming conventions followed well.

As far as I can see, no big SOLID or Clean Architecture violations.

Approving in the interest of getting our program working first, and then fixing all the smaller issues later.

**Review by yyutiw — DISMISSED** [review]

**Review by CC-3636 — DISMISSED** [review]
overall comments:
- good adherence to CA framework and SOLID principles
- good documentation
- good luck

**Comment by jltng** [comment]
request closed because created new branch and merged use case files

**Pull request closed without merging**